File: hackerrank/compare_triplets.js
function compareTriplets(scores1, scores2) {
  let scoreTotal1 = 0;
  let scoreTotal2 = 0;

  for(let i = 0; i < scores1.length; i++) {
    if(scores1[i] > scores2[i]) {
      scoreTotal1++;
    } else if (scores2[i] > scores1[i]) {
      scoreTotal2++;
    }
  }

  return [scoreTotal1, scoreTotal2];
}

//time complexity = O(n) because looping through an array once
//space complexity = O(n) because creating 2 arrays -> 2n simplifies to n

let answer = compareTriplets([5,6,7], [3,6,8]);
console.log(answer);

File: hackerrank/compare_triplets.rb
def compareTriplets(a0, a1, a2, b0, b1, b2)
  score1 = [a0, a1, a2]
  score2 = [b0, b1, b2]

  score_total_1 = 0
  score_total_2 = 0

  score1.each_index do |idx|
    if score1[idx] > score2[idx]
      score_total_1 += 1
    elsif score2[idx] > score1[idx]
      score_total_2 += 1
    end
  end

  p [score_total_1, score_total_2]
end

compareTriplets(5, 6, 7, 3, 6, 10)
